scripts/memory_analysis.py

```
@@ -0,0 +1,228 @@
+"""
+Memory usage analysis for TableConverter.
+
+This script analyzes the theoretical memory footprint of the TableConverter
+and provides recommendations for memory-safe default limits.
+"""
+
+import sys
+from typing import Dict, Any, List
+
+
+def estimate_object_memory_size(obj: Dict[str, Any]) -> int:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)